### PR TITLE
Add editorconfig file for APCu

### PR DESCRIPTION
**Pull request by TysonAndre**
https://editorconfig.org/ has integrations with many editors/IDEs to apply project-specific settings (e.g. to consistently use tabs)

https://www.php-fig.org/psr/psr-12/ recommends using spaces for php code. (though many of these tests use tabs, I don't have a strong preference)

Replace the mix of tabs and spaces in config.m4 with spaces.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+indent_style = space
+[*.xml]
+indent_size = 1
+indent_style = space
+[*.m4]
+indent_size = 2
+indent_style = space
+[*.w32]
+charset = utf-8
+end_of_line = crlf
```
Comment (nikic) [inline .editorconfig:19]: Is this really needed and not handled by git checkout?

Comment (TysonAndre) [reply .editorconfig:19]: > Is this really needed and not handled by git checkout?

I'm not sure what you mean.

`*` was added to use line feed for all other types of files (e.g. brand new files added by contributors), this is added to override that for the line endings currently used by config.w32. (`\r\n`)

Some editors/IDEs do a bad job with pasting in snippets copied from elsewhere and can end up with mixes of newline types

Comment (nikic) [reply .editorconfig:19]: Well, why is this commit changing that file? That shouldn't happen.

Comment (nikic) [reply .editorconfig:19]: Most likely end_of_line just shouldn't be specified in editorconfig at all, and left up to git.

Comment (TysonAndre) [reply .editorconfig]: https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings#per-repository-settings are you referring to this feature of git? https://github.com/krakjoe/apcu doesn't have a `.gitattributes` file.

File: config.w32 [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
+
+	/* XXX srwlock kernel as most BC compatible for now */
+	AC_DEFINE('APC_SRWLOCK_KERNEL', 1);
+	AC_DEFINE('HAVE_APCU', 1);
+	ADD_FLAG('CFLAGS_APCU', '/D WIN32_ONLY_COMPILER=1 /DAPC_SRWLOCK_KERNEL=1');
+	PHP_INSTALL_HEADERS("ext/apcu", "php_apc.h apc.h apc_api.h apc_cache.h apc_globals.h apc_iterator.h apc_lock.h apc_mutex.h apc_sma.h apc_serializer.h apc_stack.h apc_windows_srwlock_kernel.h apc_arginfo.h php_apc_legacy_arginfo.h");
+
+	EXTENSION('apcu', apc_sources, PHP_APCU_SHARED, "/DZEND_ENABLE_STATIC_TSRMLS_CACHE=1");
+}
+
```
Comment (nikic) [inline config.w32:32]: The changes to this file still need to be dropped.

File: .editorconfig (hunk 1 above)
Comment (nikic) [reply .editorconfig]: I'm referring to the general core.autocrlf setting, which people can use to convert LF to CRLF in Windows checkouts, if necessary.

File: config.w32 (hunk 2 above)
Comment (TysonAndre) [reply config.w32]: Done

File: .editorconfig (hunk 1 above)
Comment (TysonAndre) [reply .editorconfig]: Removed the end_of_line override

**Pull request merged**